File: src/main/java/org/noxet/noxetserver/playerdata/types/PDTBoolean.java
package org.noxet.noxetserver.playerdata.types;

import org.bukkit.configuration.file.YamlConfiguration;
import org.noxet.noxetserver.playerdata.PlayerDataType;

public class PDTBoolean implements PlayerDataType<Boolean> {
    @Override
    public Boolean getEmptyValue() {
        return false;
    }

    @Override
    public Boolean getValue(YamlConfiguration config, String key) {
        return config.getBoolean(key);
    }
}

File: src/main/java/org/noxet/noxetserver/playerdata/types/PDTBoolean.scala
package org.noxet.noxetserver.playerdata.types

import org.bukkit.configuration.file.YamlConfiguration
import org.noxet.noxetserver.playerdata.PlayerDataType

class PDTBoolean extends PlayerDataType[Boolean]:
  override def getEmptyValue: Boolean = false

  override def getValue(config: YamlConfiguration, key: String): Boolean =
    config.getBoolean(key)